core/src/main/java/org/apache/accumulo/core/client/admin/SummaryRetriever.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Map;
+
+import org.apache.hadoop.io.Text;
+
+public interface SummaryRetriever {
+
+  /**
+   * Allows optionally setting start row before retrieving data.
+   */
+  SummaryRetriever start(Text startRow);
```

Comment:
I was also thinking that multiple summaries could be kept per rfile. That could work something like the following.

 * After 1,000 key values are written create a Tuple\<end row, summary\>
 * Add this tuple to an in memory list of the form List\<Tuble\<endrow, summary\>> called summaryList.  
 * If summaryList exceeds 10, then merge as merge(summaryList[0],  summaryList[1]),  merge(summaryList[2],  summaryList[3]), etc... halving the size of the list.

So basically we could easily build more granular summary info within a file that, does not exceed a fixed size, by leveraging the merge functionality of a summarizer.